src/world/message_server.cpp

```diff
@@ -248,11 +297,14 @@ void message_server_listen(const bool& requestExit)
     }
 }
 
-void message_server_init(const bool& requestExit)
+void message_server_init(const std::shared_ptr<SqlConnection>&               sql,
```

Comment:
You've activated my trap card (and I think you messaged me about this).

The underlying SQL handle we use in SqlConnection is ABSOLUTELY NOT THREAD SAFE EVEN SLIGHTLY. If you access it from a different thread from where it was created it'll wipe it out silently and then the next access will crash the system. message_server lives on its own thread, so you can't pass a shared_ptr to a SqlConnection created on the main thread to it, it needs to have its own one (like I had set up before).

There's no harm in the message_server, the parent world server, and each message-handling-regional-event-thing having their own SqlConnections as part of their classes, they're very cheap and SQL Server is very performant

Reply:
I've switched to creating instances of sql connection as needed, or using the global when available.